src/api.ts

```diff
@@ -145,14 +147,18 @@ export function request<Return, Error, Body>(
     .then((res: unknown) => {
       // response will always be type 'Response'
       const response = res as Response;
+      response.headers.forEach((value, key) => responseHeaders[key] = value);
```

Comment:
Would this be similar?
```suggestion
      responseHeaders = response.headers;
```